### DISCUSSION TITLE
Help Adding FirstEnergy Support

### DISCUSSION THREAD

**Discussion by eracknaphobia** [tronikos/opower#37] [General]
I am attempting to add support for FirstEnergy which has the following electric companies:
- Ohio Edison
- Illuminating Company
- Toledo Edison
- Met-Ed
- Penelec
- Penn Power
- West Penn Power
- JCP&L
- Mon Power
- Potomac Edison

The problem is all the `r8loFrsvDB-?` values in the POST body are created by javascript as far as I can tell. I tested by disabling javascript and the post failed with these values missing.

Login URL:
https://www.firstenergycorp.com/log_in.fecorplogin.html

Login Post Example:

![image](https://github.com/tronikos/opower/assets/8268725/008bfc42-5248-41f1-83f1-48dfe0ecf764)

![image](https://github.com/tronikos/opower/assets/8268725/4e3a0b98-fd02-4d5f-8ab2-f7b536771b70)

Once logged in you can see they do make calls to opower.com, so this should be a valid provider.

![image](https://github.com/tronikos/opower/assets/8268725/e7391228-32d2-46b6-a3ed-39c26b66a52c)

I'm hoping the community could help come up with a way to get around this issue.

**Comment by tronikos** [comment]
See https://github.com/tronikos/opower/discussions/19

**Comment by EricGatesIT** [comment]
@tronikos Thanks, I guess I should've checked the closed issues before posting. Not looking good for the FirstEnergy customers. 🙁
